lib/wrap-http-client.js

```diff
@@ -19,17 +19,18 @@ import createBackoff from './create-backoff'
  * @prop {boolean} retryOnTooManyRequests - If we should retry on 429s
  */
 export default function wrapHttpClient (http, {concurrency, delay, maxRetries, retryOnTooManyRequests}) {
-  return ['get', 'post', 'put', 'delete', 'patch', 'head'].reduce(
+  const backoffClient = ['get', 'post', 'put', 'delete', 'patch', 'head'].reduce(
     (http, methodName) => {
       let httpCall = http[methodName].bind(http)
       if (retryOnTooManyRequests) {
-        httpCall = maybeBackoff(httpCall, maxRetries)
+        http[methodName] = maybeBackoff(httpCall, maxRetries)
       }
-      http[methodName] = rateLimit(httpCall, concurrency, delay)
       return http
     },
     cloneDeep(http)
   )
+
+  return rateLimit(backoffClient, ['get', 'post', 'put', 'delete', 'patch', 'head'], concurrency, delay)
 }
 
 function maybeBackoff (fn, maxRetries) {
```

Comment:
@madtrick yes that's the plan the `make-backoff` and `wrap-http-client` will be removed and `rate-limit` will be use directly